lib/iris/coords.py

```diff
@@ -497,6 +513,14 @@ def is_full_slice(s):
                 if bounds is not None:
                     bounds = bounds[keys + (Ellipsis, )]
 
+        # Copy data after indexing, if needed, to avoid making coords that are
+        # views on other coords.  This will not realise lazy data.
+        if not points_copied:
+            points = points.copy()
```

Comment:
No, because this is making copies of the new derived arrays, that will be passed as the new data to Coord.copy().
Meanwhile L552 is copying is the *old* (full) data -- which in this case will just get replaced (!!)
That is indeed a potential problem, but the existing code is already doing it :  As I've managed not to touch that code in this PR, I don't want to address it now.